cli/src/commands/config/edit.rs

```diff
@@ -40,6 +41,36 @@ pub fn cmd_config_edit(
     if !file.path().exists() {
         file.save()?;
     }
-    editor.edit_file(file.path())?;
+
+    // Editing again and again until
+    // 1. The config is OK
+    // 2. The user restore previous one
+    let mut edit_config = true;
+    while edit_config {
+        editor.edit_file(file.path())?;
+
+        // Trying to load back config. If error, prompt to continue editing
+        if let Err(e) = ConfigLayer::load_from_file(file.layer().source, file.path().to_path_buf())
+        {
+            let command_error: CommandError = e.into();
+            print_error(
+                ui,
+                "Config error: ",
+                &command_error.error,
+                &command_error.hints,
+            )?;
+            edit_config = ui.prompt_yes_no(
+                "Do you want to keep editing the file ? If not, previous config will be restored. ",
```

Comment:
I had to use print_error here so i can print precise TOML error without exiting jj. i don't know if there is a cleaner way to do this (without having print_error going `pub`)

Reply:
You can use `writeln!(ui.warning_default(), ..)` (and `print_error_sources()` if needed.)

Reply:
oh nice print_error_sources is much simpler. I used it as it allows user to directly see where the error is (even if it's pretty obvious that it's coming from what the user just added to the config)